### PR TITLE
Support GitLab in README files

### LINKED ISSUE #1223: Images in README.md of GitLab repositories not loaded
**What I did**

- Package hosted on GitLab.com
- Add README.md with screenshots / images
- Add to package to Packagist

**What I expected**

- Images are rendered in the package detail view on packagist.com (just like any package hosted on GitHub.com)

**What happend instead?**

- The image is not rendered
- The relative link is not transformed to an absolute link pointing to GitLab.com, and therefore will fail to load

**Additional information**

I searched for [existing tickets](https://github.com/composer/packagist/search?q=gitlab+images&type=issues) and this seems to be an ongoing issue, with the images from GitLab being allowed, then removed again.

The current app does look for *GitHub* Readmes and replaces image links inside of them, see https://github.com/composer/packagist/blob/787bd997f96fcb812efa2b3683acb1770269d5df/src/Package/Updater.php#L682-L693

[Example package](https://packagist.org/packages/pixelbrackets/markdown-mini-page) hosted on GitLab and shown on Packagist with missing image on top (*Screenshot*).

### PR DESCRIPTION
**Pull request by pixelbrackets**
Add support for GitLab in README files as well.

- Convert relative links to absolute links,
  pointing to the right repository subdirectory

- Fix path to images sources to embed them directly

Tested manually with [example package pixelbrackets/markdown-mini-page](https://packagist.org/packages/pixelbrackets/markdown-mini-page).

Before: [Missing screenshot](https://user-images.githubusercontent.com/1592995/152661116-5078ed64-aaff-4fb1-b889-85a07970d9a3.png)
After: [Having screenshot](https://user-images.githubusercontent.com/1592995/152661117-bdf5a2b5-f342-43d8-9768-649c1134f495.png)

As suggested in issue #1223 I tried to take a look at BitBucket as well,
but it turns out they generate hashes for all embedded image paths,
which can not be guessed (see https://bitbucket.org/pixelbrackets/composer-test/).

Closes #1223

### CODE REVIEW

**Comment by pixelbrackets** [comment]
To support other hosts than GitHub in the README [preparation method](https://github.com/composer/packagist/blob/787bd997f96fcb812efa2b3683acb1770269d5df/src/Package/Updater.php#L608) I changed the method argument `$isGitHub` to a more generic `$host`. Is this okay with you?

The fixtures have GitHub repos only. Therefore my note that I tested the change manually with an example package of my own. Is this good enough to reproduce the feature?

**Comment by Seldaek** [comment]
Thanks, looks good to me I think.

Regarding bitbucket it seems to me like the hash is simply the commit hash, and it can be replaced by the branch name or HEAD as well e.g. https://bitbucket.org/pixelbrackets/composer-test/raw/HEAD/docs/screenshot.png works fine. So I think you should be able to add bitbucket support if you still want to do that?

**Comment by Seldaek** [comment]
I'm already merging this for now though, if you wanna do bitbucket please send another PR :)

File: src/Package/Updater.php [hunk 1]
```diff
@@ -680,27 +680,29 @@ private function prepareReadme(string $readme, ?string $host = null, ?string $ow
...
+        if (in_array($host, ['github.com', 'gitlab.com'], true)) {
             // convert relative to absolute images
             $images = $dom->getElementsByTagName('img');
             foreach ($images as $img) {
-                if (false === strpos($img->getAttribute('src'), '//')) {
-                    $imgSrc = ($host === 'github.com')?
-                        'https://raw.github.com/'.$owner.'/'.$repo.'/HEAD/'.$basePath.$img->getAttribute('src') :
-                        'https://gitlab.com/'.$owner.'/'.$repo.'/-/raw/HEAD/'.$basePath.$img->getAttribute('src');
+                if (!str_contains($img->getAttribute('src'), '//')) {
+                    $imgSrc = match ($host) {
```
Comment (pixelbrackets) [inline src/Package/Updater.php:701]: Ah, thats much better readable. Cool :+1:

**Comment by pixelbrackets** [comment]
> Thanks, looks good to me I think.
> 
> Regarding bitbucket it seems to me like the hash is simply the commit hash, and it can be replaced by the branch name or HEAD as well e.g. https://bitbucket.org/pixelbrackets/composer-test/raw/HEAD/docs/screenshot.png works fine. So I think you should be able to add bitbucket support if you still want to do that?

Hah, thanks for the clue. I'll pick that up in another PR.

**Pull request merged**